src/compile/scale/type.ts

```diff
@@ -16,56 +16,58 @@ export type RangeType = 'continuous' | 'discrete' | 'flexible' | undefined;
  */
 // NOTE: CompassQL uses this method.
 export default function type(
-  specifiedType: ScaleType, type: Type, channel: Channel, timeUnit: TimeUnit, mark: Mark,
+  specifiedType: ScaleType, channel: Channel, fieldDef: FieldDef, mark: Mark,
   hasTopLevelSize: boolean, specifiedRangeStep: number, scaleConfig: ScaleConfig): ScaleType {
 
+  const defaultScaleType = defaultType(channel, fieldDef, mark, hasTopLevelSize, specifiedRangeStep, scaleConfig);
```

Comment:
It's not really a big deal, but I think it's worth mentioning that my original code only runs defaultType if necessary.  Currently code always run it, so it's a little more wasteful, but I'm okay with this too, since it's a very cheap method to run.